src/circuit/circuit_vanilla/gadget.rs

```diff
@@ -1,190 +1,6 @@
-        value_commit_v.mul(layouter.namespace(|| "[v] ValueCommitV"), v)?
-    };
-
-    // blind = [rcv] ValueCommitR
-    let (blind, _rcv) = {
-        let value_commit_r = OrchardFixedBasesFull::ValueCommitR;
-        let value_commit_r = FixedPoint::from_inner(ecc_chip, value_commit_r);
-
-        // [rcv] ValueCommitR
-        value_commit_r.mul(layouter.namespace(|| "[rcv] ValueCommitR"), rcv)?
-    };
-
-    // [v] ValueCommitV + [rcv] ValueCommitR
-    commitment.add(layouter.namespace(|| "cv"), &blind)
-}
-
-/// `DeriveNullifier` from [Section 4.16: Note Commitments and Nullifiers].
-///
-/// [Section 4.16: Note Commitments and Nullifiers]: https://zips.z.cash/protocol/protocol.pdf#commitmentsandnullifiers
-#[allow(clippy::too_many_arguments)]
-pub(in crate::circuit) fn derive_nullifier<
-    PoseidonChip: PoseidonSpongeInstructions<pallas::Base, poseidon::P128Pow5T3, ConstantLength<2>, 3, 2>,
-    AddChip: AddInstruction<pallas::Base>,
-    EccChip: EccInstructions<
-        pallas::Affine,
-        FixedPoints = OrchardFixedBases,
-        Var = AssignedCell<pallas::Base, pallas::Base>,
-    >,
->(
-    mut layouter: impl Layouter<pallas::Base>,
-    poseidon_chip: PoseidonChip,
-    add_chip: AddChip,
-    ecc_chip: EccChip,
-    rho: AssignedCell<pallas::Base, pallas::Base>,
-    psi: &AssignedCell<pallas::Base, pallas::Base>,
-    cm: &Point<pallas::Affine, EccChip>,
-    nk: AssignedCell<pallas::Base, pallas::Base>,
-) -> Result<X<pallas::Affine, EccChip>, plonk::Error> {
-    // hash = poseidon_hash(nk, rho)
-    let hash = {
-        let poseidon_message = [nk, rho];
-        let poseidon_hasher =
-            PoseidonHash::init(poseidon_chip, layouter.namespace(|| "Poseidon init"))?;
-        poseidon_hasher.hash(
-            layouter.namespace(|| "Poseidon hash (nk, rho)"),
-            poseidon_message,
-        )?
-    };
-
-    // Add hash output to psi.
-    // `scalar` = poseidon_hash(nk, rho) + psi.
-    let scalar = add_chip.add(
-        layouter.namespace(|| "scalar = poseidon_hash(nk, rho) + psi"),
-        &hash,
-        psi,
-    )?;
-
-    // Multiply scalar by NullifierK
-    // `product` = [poseidon_hash(nk, rho) + psi] NullifierK.
-    //
-    let product = {
-        let nullifier_k = FixedPointBaseField::from_inner(ecc_chip, NullifierK);
-        nullifier_k.mul(
-            layouter.namespace(|| "[poseidon_output + psi] NullifierK"),
-            scalar,
-        )?
-    };
-
-    // Add cm to multiplied fixed base to get nf
-    // cm + [poseidon_output + psi] NullifierK
-    cm.add(layouter.namespace(|| "nf"), &product)
-        .map(|res| res.extract_p())
-}
-
 pub(in crate::circuit) use super::commit_ivk::gadgets::commit_ivk;
+pub(in crate::circuit) use super::derive_nullifier::gadgets::derive_nullifier;
 pub(in crate::circuit) use super::note_commit::gadgets::note_commit;
+pub(in crate::circuit) use super::value_commit_orchard::gadgets::value_commit_orchard;
```

Comment:
can we remove this file completely? (x2)

Reply:
I moved all logic from circuit_vanilla/value_commit_orchard.rs and circuit_zsa/value_commit_orchard.rs to circuit/value_commit_orchard.rs